libs/wingsdk/src/target-sim/function.inflight.ts

```diff
@@ -116,31 +112,38 @@ async function runInSandbox(code: string, payload: any, opts: RunCodeOptions) {
   // we are hijacking console.log to log to the inflight $logger so do not propagate
   delete ctx.console;
 
-  return new Promise(($resolve, $reject) => {
-    const wrapper = [
-      "const exports = {};",
-      "Object.assign(process.env, $env);",
-      code,
-      // The last statement is the value that will be returned by vm.runInThisContext
-      `exports.handler(${JSON.stringify(
-        payload
-      )}).then($resolve).catch($reject);`,
-    ].join("\n");
-
-    // we want "require"s to resolve relative to the directory of the user's code
-    const requireResolve = (p: string) =>
-      require.resolve(p, { paths: [opts.resolveDir] });
-    // eslint-disable-next-line @typescript-eslint/no-require-imports
-    const inflightRequire = (p: string) => require(requireResolve(p));
-    inflightRequire.resolve = requireResolve;
-
-    const context = vm.createContext({
-      ...ctx,
-      require: inflightRequire,
-      $resolve,
-      $reject,
-    });
+  // we want to intercept "require" calls and execute the code in the same context so that we can
+  // share state between modules within the same inflight VM.
+  const makeRequire = (parent: string) =>  {
+
+    const customResolve = (request: string) => {
+      return require.resolve(request, { paths: [dirname(parent)] });
+    };
+
+    const customRequire = (request: string) => {
+      if (Module.isBuiltin(request)) {
+        return require(request);
+      }
+
+      const filename = customResolve(request);
+      const code = fs.readFileSync(filename, "utf-8");
+      
+      // wrap the module based on https://nodejs.org/api/modules.html#the-module-wrapper
+      const wrapped = `(function(exports, require, module, __filename, __dirname) {${code}});`
```

Comment:
Can we just emit `require(${filename})` so that the entrypoint will behave like any other module?